src/Chapter4.hs

```diff
@@ -629,7 +646,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM l r = l >>= (\lres -> if lres then r >>= (\rres -> pure $ lres && rres) else pure lres)
```

Comment:
You need to just return `r` in case of False, then the implementation is going to be lazy, as needed 🙂 
```suggestion
andM l r = l >>= \lres -> if lres then r else pure False
```